### PR TITLE
dog: add page

### PR DESCRIPTION
**Pull request by jriggins**
- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):**
v0.1.0

Note I just pulled the examples from the dog man page (as I'm still trying to learn how to use it myself 😅 )

/cc @ogham I believe the creator of [dog](https://dns.lookup.dog/)

### CODE REVIEW

File: pages/common/dog.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# dog
+
+> Dog is a command-line DNS client.  It has colourful output, supports the DNS-over-TLS and DNS-over-HTTPS protocols, and can emit JSON.
+> More information: <https://dns.lookup.dog/>.
+
+- Query the A record of a domain using default settings:
```
Comment (navarroaxel) [inline pages/common/dog.md:6]: This is more clear for newbies
```suggestion
- Lookup the IP(s) associated with a hostname (A records):
```

File: pages/common/dog.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+`dog example.net`
+
+- Look up MX records instead:
```
Comment (navarroaxel) [inline pages/common/dog.md:10]:
```suggestion
- Query MX records type associated with a given domain name:
```

File: pages/common/dog.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+`dog example.net MX`
```
Comment (navarroaxel) [inline pages/common/dog.md:12]:
```suggestion
`dog {{example.com}} MX`
```

File: pages/common/dog.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+- Query using a specific nameserver instead:
```
Comment (navarroaxel) [inline pages/common/dog.md:14]:
```suggestion
- Specify a specific DNS server to query (e.g. Cloudflare):
```

File: pages/common/dog.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+`dog example.net MX @1.1.1.1`
+
+- Query using TCP rather than UDP:
+
+`dog example.net MX @1.1.1.1 -T`
+
+- As above, but using explicit arguments:
+
+`dog -q example.net -t MX -n 1.1.1.1 -T`
```
Comment (navarroaxel) [inline pages/common/dog.md:24]: We prefer long arguments
```suggestion
`dog -q {{example.com}} --type MX --nameserver {{1.1.1.1}} --tcp`
```

File: pages/common/dog.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+`dog example.net MX`
+
+- Query using a specific nameserver instead:
+
+`dog example.net MX @1.1.1.1`
+
+- Query using TCP rather than UDP:
+
+`dog example.net MX @1.1.1.1 -T`
```
Comment (navarroaxel) [inline pages/common/dog.md:20]:
```suggestion
`dog example.net MX @{{1.1.1.1}} --tcp`
```

**Comment by navarroaxel** [comment]
:thinking: I wonder if we can add specific examples above how to force DoH using `1dot1dot1dot1.cloudflare-dns.com` and JSON output

File: pages/common/dog.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+`dog {{example.com}} MX`
+
+- Specify a specific DNS server to query (e.g. Cloudflare):
+
+`dog {{example.com}} MX @{{1.1.1.1}}`
+
+- Query using TCP rather than UDP:
+
+`dog example.net MX @{{1.1.1.1}} --tcp`
```
Comment (navarroaxel) [inline pages/common/dog.md:21]:
```suggestion
`dog {{example.com}} MX @{{1.1.1.1}} --tcp`
```

**Comment by jriggins** [comment]
> 1dot1dot1dot1.cloudflare-dns.com

@navarroaxel How about this example taken from [here](https://dns.lookup.dog/features/dns-over-https)?

```bash
dog example.com --https @https://cloudflare-dns.com/dns-query
```

**Comment by navarroaxel** [comment]
> @navarroaxel How about this example taken from [here](https://dns.lookup.dog/features/dns-over-https)?
> 
> ```shell
> dog example.com --https @https://cloudflare-dns.com/dns-query
> ```

@jriggins sounds good

**Review by EmilySeville7cfg — APPROVED** [review]

File: pages/common/dog.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+- Query over TCP rather than UDP:
+
+`dog {{example.com}} MX @{{1.1.1.1}} --tcp`
+
+- Query the MX records type associated with a given domain name over TCP using explicit arguments:
+
+`dog --query {{example.com}} --type MX --nameserver {{1.1.1.1}} --tcp`
+
+- Query over DNS-over-HTTPS (DoH):
```
Comment (navarroaxel) [inline pages/common/dog.md:27]:
```suggestion
- Lookup the IP(s) associated with a hostname (A records) using DNS over HTTPS (DoH):
```

File: pages/common/dog.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+`dog {{example.com}} --https @https://cloudflare-dns.com/dns-query`
```
Comment (navarroaxel) [inline pages/common/dog.md:29]:
```suggestion
`dog {{example.com}} --https @{{https://cloudflare-dns.com/dns-query}}`
```

**Review by navarroaxel — APPROVED** [review]

**Comment by jriggins** [comment]
Thanks for all of your help @navarroaxel and @EmilySeville7cfg!

**Pull request merged**